File: Part 2 - Regression/Section 5 - Multiple Linear Regression/Multiple_Linear_Regression.py
# Multiple Linear Regression

# Importing the Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the Dataset
dataset = pd.read_csv('50_Startups.csv')
X = dataset.iloc[:,:-1].values
Y = dataset.iloc[:,4].values
np.set_printoptions(threshold = np.nan)  # If you cant see full array list

# Encoding Categorical Data
# Encoding Independent Variable
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
labelencoder_X = LabelEncoder()
X[:,3] = labelencoder_X.fit_transform(X[:,3])
onehotencoder = OneHotEncoder(categorical_features = [3])
X = onehotencoder.fit_transform(X).toarray()

# Avoiding the Dummy Variable Trap
X = X[:,1:] 

# Splitting the Dataset into Training set & Test Set
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X,Y, test_size = 0.2, random_state = 0)

# Fitting Multiple Linear Regression to the Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)

# Predicting the Test Set Results
Y_pred = regressor.predict(X_test)

File: Part 2 - Regression/Section 5 - Multiple Linear Regression/Multiple_Linear_Regression.R
# Multiple Linear Regression

# Simple Linear Regression

# Data Preprocessing

# Importing the Dataset
dataset = read.csv('50_Startups.csv')
#dataset = dataset[ ,2:3]

# Encoding Categorical Data
dataset$State = factor(dataset$State,
                         levels = c('New York','California','Florida'),
                         labels = c(1,2,3))
 

# Splitting the Dataset into Training set and Test set
# install.packages('caTools')  - To install a package
library(caTools)   #- To select the library in manual way
set.seed(123)
split = sample.split(dataset$Profit, SplitRatio = 0.8)
training_set = subset(dataset,split == TRUE)
test_set = subset(dataset,split == FALSE) 

# Feature Scaling
# training_set[ , 2:3] = scale(training_set[ , 2:3])
# test_set[ , 2:3] = scale(test_set[ , 2:3])

# Fitting Multiple Linear Regression to the Training Set
regressor = lm(formula = Profit ~ R.D.Spend + Administration + Marketing.Spend + State, # or (formula = profit ~ .)
               data =  training_set)

# Predicting the Test Set Result
y_pred = predict(regressor, newdata = test_set)

# Building Optimal Model using Backward Elimination
regressor = lm(formula = Profit ~ R.D.Spend + Administration + Marketing.Spend + State, # or (formula = profit ~ .)
               data =  dataset)
summary(regressor)

# Removing Features for getting Optimal Model using Backward Elimination
regressor = lm(formula = Profit ~ R.D.Spend + Administration + Marketing.Spend, # or (formula = profit ~ .)
               data =  dataset)
summary(regressor)

# Removing Features for getting Optimal Model using Backward Elimination
regressor = lm(formula = Profit ~ R.D.Spend +  Marketing.Spend, # or (formula = profit ~ .)
               data =  dataset)
summary(regressor)

# Removing Features for getting Optimal Model using Backward Elimination
regressor = lm(formula = Profit ~ R.D.Spend , # or (formula = profit ~ .)
               data =  dataset)
summary(regressor)
 

